api/v1alpha3/azuremachine_validation.go

```diff
@@ -51,3 +53,38 @@ func ValidateUserAssignedIdentity(identityType VMIdentity, userAssignedIdentetie
 
 	return allErrs
 }
+
+// ValidateOSDisk validates the OSDisk spec
+func ValidateOSDisk(osDisk OSDisk, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if osDisk.DiskSizeGB <= 0 || osDisk.DiskSizeGB > 2048 {
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("DiskSizeGB"), "", "the Disk size should be a value between 1 and 2048"))
+	}
+
+	if osDisk.OSType == "" {
+		allErrs = append(allErrs, field.Required(fieldPath.Child("OSType"), "the OS type cannot be empty"))
+	}
+
+	allErrs = append(allErrs, validateStorageAccountType(osDisk.ManagedDisk.StorageAccountType, fieldPath)...)
+
+	return allErrs
+}
+
+func validateStorageAccountType(storageAccountType string, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	storageAccTypeChildPath := fieldPath.Child("ManagedDisk").Child("StorageAccountType")
+
+	if storageAccountType == "" {
+		allErrs = append(allErrs, field.Required(storageAccTypeChildPath, "the Storage Account Type for Managed Disk cannot be empty"))
+		return allErrs
+	}
+
+	for _, possibleStorageAccountType := range compute.PossibleDiskStorageAccountTypesValues() {
```

Comment:
@devigned do you foresee any issues with taking a dependency on the SDK in the CAPZ api package for k8s-infra integration?

Reply:
maybe this sort of validation would move to k8s-infra webhooks eventually, but for now it allows failing fast if the user specifies known invalid values.

Reply:
The way the infra controller works now, get request to the API for skus and related look up data isn’t implemented. It probably should be, and I’ll need to think about that.

As for quick validation, can’t argue with the effectiveness of having that check here.